flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarOperators.scala

```diff
@@ -911,6 +911,64 @@ object ScalarOperators {
     }
   }
 
+  def generateRand(
+    randField: String,
+    seedExpr: GeneratedExpression,
+    resultType: TypeInformation[_])
+  : GeneratedExpression = {
+    val resultTerm = newName("result")
+    val resultTypeTerm = primitiveTypeTermForTypeInfo(resultType)
+    val randCode = if (seedExpr != null) {
+      s"""
+         |if ($randField == null) {
+         |  ${seedExpr.code}
+         |  $randField = new java.util.Random(${seedExpr.resultTerm});
```

Comment:
What happens if `seedExpr.nullTerm` is true? We need a case distinction for both null check enabled and disabled.